### PR TITLE
generate systemd

### PR DESCRIPTION
**Pull request by vrothberg**
Implement `podman generate systemd` for Podman v2 and enable associated
tests.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
Generation is working but there are still some rough edges when trying to run container in systemd. I also noticed that the system tests need some attention but deferred that to later.

**Comment by vrothberg** [comment]
@giuseppe @rhatdan PTAL

**Comment by rhatdan** [comment]
LGTM

File: cmd/podman/generate/generate.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"github.com/containers/libpod/pkg/util"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// Command: podman _generate_
+	generateCmd = &cobra.Command{
+		Use:              "generate",
+		Short:            "Generate structured data based on containers and pods.",
+		Long:             "Generate structured data (e.g., Kubernetetes yaml or systemd units) based on containers and pods.",
```
Comment (TomSweeneyRedHat) [inline cmd/podman/generate/generate.go:15]: Kubernetetes -> Kubernetes

Comment (vrothberg) [reply cmd/podman/generate/generate.go]: Nice catch!

**Comment by vrothberg** [comment]
Rebased: @baude @giuseppe @jwhonce @mheon PTAL

File: cmd/podman/generate/generate.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		Long:             "Generate structured data (e.g., Kubernetes yaml or systemd units) based on containers and pods.",
+		TraverseChildren: true,
+		RunE:             registry.SubCommandExists,
+	}
+	containerConfig = util.DefaultContainerConfig()
+)
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode, entities.TunnelMode},
```
Comment (jwhonce) [inline cmd/podman/generate/generate.go:24]:
```suggestion
		Mode:    []entities.EngineMode{entities.ABIMode},
```

**Comment by vrothberg** [comment]
> Small change if command is not to be supported in tunnel mode

Updated, thanks!

**Pull request merged**